mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -551,12 +553,17 @@ class Word2VecModel private[spark] (
       }
       ind += 1
     }
-    wordList.zip(cosVec)
+    var topResults = wordList.zip(cosVec)
       .toSeq
-      .sortBy(- _._2)
+      .sortBy(-_._2)
       .take(num + 1)
       .tail
-      .toArray
+    if (vecNorm != 0.0f) {
+      topResults = topResults.map {
+        case (word: String, cosVec: Double) => (word, cosVec / vecNorm)
```

Comment:
same here for style, prefer 
```
topResults = topResults.map { case (word: String, cosVec: Double) =>
  ...
```

(as long as it fits within line length restriction)